### ISSUE TITLE
Save access_token in users table?

### ISSUE THREAD

**Issue by natenrb9** [SammyK/LaravelFacebookSdk#29]
I have a user logging in fine and saving all of their info (id, name, email, etc) to the DB, but it does not seem to be saving the access_token? Isn't it supposed to be? I am not sure why it isn't saving that portion in the DB...I can't tell if it is a bug, or something I need to configure so that it saves correctly.

**Comment by jagdeepsinghmundi** [comment]
You'll need to save it separately.

When you receive user's information from Facebook and save it using Model, it does not contain access_token in response, that is why it isn't saving the access token.

But you can do it this way...

```php
// update access token in user's profile
$user->access_token = $token;
$user->save();
```

**Comment by natenrb9** [comment]
I am using the example that had been give: https://github.com/SammyK/LaravelFacebookSdk#user-authentication-example

After reading through: https://www.sammyk.me/best-practice-for-facebook-login-with-the-javascript-sdk-and-php-sdk-v4-1

It looks like I do not need to save the access_token unless I am accessing their data when they are offline, which I am not.

REMAINING QUESTIONS:
1. What I am not understanding is where the access token is actually saved when I use the above example? Is it being saved into the JS or the PHP session? I var dumped the session data and it did not seem to be in there...

2. I can get a user to log in, but when I try to make Graph requests while they are logged in it throws this error: `Method Facebook\Entities\AccessToken::__toString() must return a string value`
I am not sure where it is trying to pull the access_token from, so it is difficult to troubleshoot and it seems like the code should already convert it over. This is the code I am trying to execute in my controller:
`$friends = Facebook::edge('user_friends')->fields('id', 'name')->limit(5);
$user = Facebook::object('me')->fields('name', $friends)->get();`

**Comment by natenrb9** [comment]
I am using the example that had been give: https://github.com/SammyK/LaravelFacebookSdk#user-authentication-example

After reading through: https://www.sammyk.me/best-practice-for-facebook-login-with-the-javascript-sdk-and-php-sdk-v4-1

It looks like I do not need to save the access_token unless I am accessing their data when they are offline, which I am not.

REMAINING QUESTIONS:
-1. What I am not understanding is where the access token is actually saved when I use the above example? Is it being saved into the JS or the PHP session? I var dumped the session data and it did not seem to be in there...

-2. I can get a user to log in, but when I try to make Graph requests while they are logged in it throws this error: `Method Facebook\Entities\AccessToken::__toString() must return a string value`

I am not sure where it is trying to pull the access_token from, so it is difficult to troubleshoot and it seems like the code should already convert it over. 

This is the code I am trying to execute in my controller:
`$friends = Facebook::edge('user_friends')->fields('id', 'name')->limit(5);
$user = Facebook::object('me')->fields('name', $friends)->get();`

**Comment by natenrb9** [comment]
Ok, I figured it out, looks like I needed to add something in to actually set the token to the session:

`Session::put('FacebookAccessToken', (string) $token);`

And then when making a request, I needed to retrieve it from the session and re-set it:

`$token = Session::get('FacebookAccessToken');
Facebook::setAccessToken($token);`

This wasn't super clear in the info give, I would maybe update to make this part clearer. It seemed like this was take care of by the package when setting the token was implemented. I read through the README.md of both SammyK/LaravelFacebookSdk and SammyK/FacebookQueryBuilder and this was not clear at all, I was only able to figure it out after reading this issue: https://github.com/SammyK/LaravelFacebookSdk/issues/24

It could be that this would be super obvious to most developers attempting to work with the SDK, but I would recommend updating the description to include info on this.

Thank you everyone for your work on getting this SDK together!
